lib/rubocop/cop/lint/unneeded_splat_expansion.rb

```diff
@@ -63,12 +63,10 @@ class UnneededSplatExpansion < Cop
           (splat {$({str dstr int float array} ...) (block #{ARRAY_NEW_PATTERN} ...) #{ARRAY_NEW_PATTERN}} ...)
         PATTERN
 
-        def on_splat(node)
-          literal_expansion?(node) do |object|
-            if object.send_type?
-              return unless ASSIGNMENT_TYPES.include?(node.parent.parent.type)
-            end
+        def_node_matcher :array_new?, ARRAY_NEW_PATTERN
```

Comment:
With this change, I would inline the pattern in `#array_new?` and call refer to that within `#literal_expansion?` instead. 🙂 (Might need to reorder the two pattern definitions.)